### ISSUE TITLE
Improve parsing of md5s/sha256s and imphashes/authentihashes

### ISSUE THREAD

**Issue by fhightower** [fhightower/ioc-finder#231]
Currently, when given an input like:

```
imphash 18ddf28a71089acdbab5038f58044c0a
```

`18ddf28a71089acdbab5038f58044c0a` will be parsed as an imphash if `parse_imphashes` is True, but will be parsed as an md5 if `parse_imphashes` is False (and the same is true of authentihashes and sha256s). To be consistent with the way similar situations are handled in this library (e.g. parsing addresses from cidr ranges), we should update the library to not parse `18ddf28a71089acdbab5038f58044c0a` from the string above even if `parse_imphashes` is False.

In other words, we should update this library to remove imphashes and authentihashes regardless of whether the commandline args (`parse_imphashes` and `parse_authentihashes`) are true.

**Comment by fhightower** [comment]
Fixed w/ #219 .
